javabasics/src/main/java/javabasics/Library.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package javabasics;
+
+import java.util.Random;
+
+
+public class Library {
+    public boolean someLibraryMethod() {
+        return true;
+    }
+
+    public int[] roll(int n){
+
+        int[] rollResult = new int[n];
+
+        for (int i = 0; i < n; i++) {
+
+            Random generator = new Random();
+            Float randomNumber = generator.nextFloat();
+
+            //This feels like the wrong way to get a number between 1 and 6, but it works, so...
+            int randomDice = Math.round(randomNumber * 5) + 1;
+
+            rollResult[i] = randomDice;
+        }
+        return rollResult;
+    }
+
+    public boolean containsDuplicates(String[] arrayToCheck) {
+        boolean hasDups = false;
+        dupCheck:
+
+            for (int i = 0; i < arrayToCheck.length; i++){
+                for (int j = 0; j < arrayToCheck.length; j++) {
+                    if ((arrayToCheck[i].equals(arrayToCheck[j])) && (i != j)) {
+                        hasDups = true;
+                        break dupCheck;
+                    }
+                }
+            }
+            return hasDups;
+    }
+
+    public int calculateAverage(int[] arrayToAverage){
+
+        if (arrayToAverage.length == 1) {
+            return arrayToAverage.length;
```

Comment:
In this conditional, you would always return 1. If you have an array with just one element, you would actually want to return the value of that element, like this: `return arrayToAverage[0];`